pootle/apps/pootle_app/views/index/index.py

```diff
@@ -7,23 +7,66 @@
 # AUTHORS file for copyright and authorship information.
 
 from django.contrib.auth import REDIRECT_FIELD_NAME
-from django.shortcuts import redirect, render
+from django.shortcuts import redirect
 from django.urls import reverse
+from django.utils.functional import cached_property
+from django.utils.translation import get_language
 from django.views.generic import TemplateView
 
+from pootle.core.decorators import persistent_property
+from pootle.core.delegate import revision, scores
 from pootle.i18n.override import get_lang_from_http_header
 from pootle_language.models import Language
+from pootle_project.models import Project, ProjectSet
 
 
 COOKIE_NAME = 'pootle-language'
 
 
+class WelcomeView(TemplateView):
+    ns = "pootle.web.welcome"
+    template_name = "welcome.html"
+
+    @property
+    def revision(self):
+        return revision.get(self.project_set.directory.__class__)(
+            self.project_set.directory).get(key="stats")
+
+    @property
+    def cache_key(self):
+        return (
+            "%s.%s.%s"
+            % (self.request.user.username,
+               self.revision,
+               self.request_lang))
+
+    @cached_property
+    def project_set(self):
+        user_projects = Project.accessible_by_user(self.request.user)
+        user_projects = (
+            Project.objects.for_user(self.request.user)
+                           .filter(code__in=user_projects))
```

Comment:
It is not the first time I see this pattern. I do wonder if it is better to move this to a new method in the manager.